### ISSUE TITLE
Training code/data release plans + a last-contig bug in the pileup indexer

### ISSUE THREAD

**Issue by virsgandhi20** [bioinfo-biols/ORCA#5]
Thanks for releasing ORCA and the pretrained models, they have been very useful.

We are using ORCA as a baseline for nanopore modification detection on our own datasets, including DNA modifications. Two questions and one bug report.

1. Training code. Are you planning to release the training scripts, in particular the domain adversarial setup? The GRL lambda schedule, learning rate, batch size, epochs, and domain label assignment would help us reproduce the leave-one-modification-out results faithfully. We reimplemented the architecture from Models_20.py and see substantially lower held-out performance than the paper reports, and in our hands the adversarial strength matters a great deal: at full GRL strength the shared embedding collapses to an effective rank of about 3 out of 640 and results become unstable across seeds, while weaker settings recover. Knowing your training hyperparameters would let us separate implementation differences from data differences.

2. Training data. Is the training data, or the processed feature tables, available anywhere? The paper does not mention a release plan.

3. Bug: the last contig of every pileup is dropped by the basecalling feature indexer. In orca/scripts/prediction_bascal_prep.py, index_pileup only writes a contig's byte range when it encounters the next contig, and never flushes after the loop ends. The final contig of every pileup is therefore missing from the basecalling features and from the merged output. For a single-contig reference (any bacterial or viral genome) this drops 100% of sites: we ran a 132 kb phage genome and the merged output contained 1 row. On our 32-construct synthetic reference the last construct was silently absent (31/32 present).

Minimal fix in index_pileup, after the loop:

if old_id != '':
    fa.write(f'{old_id},{start},{end}\n')

Workaround for users until then: append a dummy line with a fake contig name to the pileup before running orca-pred_bascal_feature_ext; the fake contig forces the flush and is never itself processed.

Happy to open a PR for the fix if useful.
